clang-tools-extra/clangd/InlayHints.cpp

```diff
@@ -651,16 +651,12 @@ class InlayHintVisitor : public RecursiveASTVisitor<InlayHintVisitor> {
     // implied object argument ([over.call.func]), the list of provided
     // arguments is preceded by the implied object argument for the purposes of
     // this correspondence...
-    //
-    // However, we don't have the implied object argument
-    // for static operator() per clang::Sema::BuildCallToObjectOfClassType.
     llvm::ArrayRef<const Expr *> Args = {E->getArgs(), E->getNumArgs()};
     // We don't have the implied object argument through a function pointer
     // either.
     if (const CXXMethodDecl *Method =
             dyn_cast_or_null<CXXMethodDecl>(Callee.Decl))
-      if (Method->isInstance() &&
-          (IsFunctor || Method->hasCXXExplicitFunctionObjectParameter()))
+      if (IsFunctor || Method->hasCXXExplicitFunctionObjectParameter())
```

Comment:
I don;t understand where the change is. `isInstance` is implied by `hasCXXExplicitFunctionObjectParameter` so isn't that change doing... nothing?

Reply:
`IsFunctor` can be true when `isInstance()` is false, this is the situation (static functor) we are dealing with.

Reply:
See https://github.com/llvm/llvm-project/pull/68485/files#diff-19c518dbc68b30c66e1a2b6bd523c005fb2050dcf1a0e92305df7ab3e1b9e9f3L15744-L15751. The patch actually changes the behavior when we put an implied object argument. (I was replying at wrong place in that PR...)

Reply:
So we may have an implied object argument for *static* operator() calls without an explicit object parameter, while it was only applied previously to *non-static* (`isInstance()`) operator() calls.